pkg/compact/compact.go

```diff
@@ -470,6 +474,70 @@ func (cg *Group) Resolution() int64 {
 	return cg.resolution
 }
 
+// should return the results/metrics of the planning simulation
+type PlanSim interface {
+	Simulate(ctx context.Context, g *DefaultGrouper, p *largeTotalIndexSizeFilter, sy *Syncer) error
+}
+
+type DefaultPlanSim struct {
+	grouper *DefaultGrouper
+	planner *largeTotalIndexSizeFilter
+	sy      *Syncer
+}
+
+func NewDefaultPlanSim(grouper *DefaultGrouper, planner *largeTotalIndexSizeFilter, sy *Syncer) *DefaultPlanSim {
+	return &DefaultPlanSim{
+		grouper: grouper,
+		planner: planner,
+		sy:      sy,
+	}
+}
+
+func (ps *DefaultPlanSim) Simulate(ctx context.Context) (int, int, error) {
```

Comment:
Let's think about the method name more. 
`Simulate` is accurate but it is not specific enough. What are we simulating for? 😄 
Maybe `ProgressSimulate` or `ProgressCalculate`?

Reply:
Agreed. Changed it to `PlanProgressCalc`. 
Does that represent the purpose accurately?